### DISCUSSION TITLE
Delete a record using Prisma client with RelationFilter in where

### DISCUSSION THREAD

**Discussion by akashrajum7** [prisma/prisma#14502] [Q&A] [answered]
Hey, 

So I have two models related to each other like so -- 

```prisma
model Profile {
  id            Int      @id @default(autoincrement())
  createdAt     DateTime @default(now())
  supertokensId String   @unique
  name          String
  tabs          Tab[]
}

model Tab {
  id            Int     @id @default(autoincrement())
  name          String
  profile       Profile @relation(fields: [supertokensId], references: [supertokensId])
  supertokensId String
}
```

Now I want to delete a Tab only if it belongs to an user, so I thought I would be able to do 

```javascript
this.prisma.tab.delete({
        where: {
          id: id,
          profile: {
            is: {
              supertokensId: userId,
            },
          },
        },
      });
```

But I get an error when I try to do this as `where` in delete can only accept ` {id: id,}`. Is there a work around for this other than having to fetch the record fist and comparing the `supertokensId` with `userId` and then using delete? I would like to avoid two database calls if possible.

Thank you.

**Comment by austincrim** [comment, marked as answer]
Hey @akashrajum7 👋,

Prisma only allows you to specify properties in `where` if they are **unique**. I think the solution here is to use a compound unique index between the `Tab` `id` and `supertokensId`.

```prisma
model Tab {
  id            Int     @id @default(autoincrement())
  name          String
  profile       Profile @relation(fields: [supertokensId], references: [supertokensId])
  supertokensId String

  @@unique([id, supertokensId])
}
```

Then in your code you can specify both:
```ts
this.prisma.tab.delete({
  where: {
    id_supertokensId: {
      id,
      supertokensId: userId
    }
  }
});
```

**Reply by akashrajum7** [reply]
Thank you! I will try this.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
